client/packages/system/src/Encounter/DetailView/DetailView.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { FC } from 'react';
+import {
+  useTranslation,
+  DetailViewSkeleton,
+  AlertModal,
+  useNavigate,
+  RouteBuilder,
+  useDebounceCallback,
+} from '@openmsupply-client/common';
+import { EncounterFragment, useEncounter } from '../api';
+
+import { useJsonForms } from '../../Patient/JsonForms';
+import { AppRoute } from '@openmsupply-client/config';
+import { Toolbar } from './Toolbar';
+
+export const DetailView: FC = () => {
+  const t = useTranslation('patients');
+  const { data: encounter, isLoading } = useEncounter.document.get();
+  const navigate = useNavigate();
+
+  const handleSave = useEncounter.document.upsert(
+    encounter?.patient.id ?? '',
+    encounter?.program ?? '',
+    encounter?.type ?? ''
+  );
+  const { JsonForm /* saveData, isDirty, validationError */ } = useJsonForms(
+    encounter?.document?.name,
+    {
+      handleSave,
+    },
+    undefined
+  );
+
+  const updateEncounter = useDebounceCallback(
+    (patch: Partial<EncounterFragment>) =>
+      handleSave?.(
+        { ...encounter?.document.data, ...patch },
+        encounter?.document?.documentRegistry?.formSchemaId ?? '',
+        encounter?.document.id
+      ),
+    [encounter],
+    1000
+  );
```

Comment:
We would  need a save button at some point, similar to patient view. For example, a lab report could be added to an existing encounter.